packages/frontend-v2/hooks/useLocalStorage.ts

```diff
@@ -1,33 +1,38 @@
-import { useState, Dispatch, SetStateAction } from "react";
+import { useState, Dispatch, useEffect } from "react";
 
-// NextJS executes server side code first, and then client side. Window object is only present client side,
-// so window could be undefined. This ensures the window object exists before we use it.
-const isWindow = typeof window !== "undefined";
+// Enum used for formik data, these entries can be deleted later.
+// These are here for testing purposes.
+export enum LocalStorageKey {
+  token = "token"
+}
 
+/**
+ * Using this hook allows for any component to access localstorage.
+ * Accessing the hook allows for the functionality of setting and getting a value in localstorage.
+ * @param key Key for the localstorage value.
+ * @param defaultValue Default value returned if the hook cannot access the value in localstorage.
+ */
 export function useLocalStorage<T>(
-  key: string,
-  initialValue: T
-): [T, Dispatch<SetStateAction<T>>] {
-  const [storedValue, setStoredValue] = useState<T>(() => {
+  key: LocalStorageKey,
+  defaultValue: T
+): [T, Dispatch<T>] {
+  const [storedValue, setStoredValue] = useState<T>(defaultValue);
+
+  useEffect(() => {
     try {
-      const item = isWindow && window.localStorage.getItem(key);
-      return item ? JSON.parse(item) : initialValue;
+      const item = window.localStorage.getItem(key);
+      setStoredValue(item ? JSON.parse(item) : defaultValue);
```

Comment:
```suggestion
      item ?? setStoredValue(JSON.parse(item));
```

no need to re-set value as defaultValue, since it will already be default value